### PR TITLE
properly drain gateway listeners

### PR DESCRIPTION
**Pull request by ramaraochavali**
As gateway listeners are created with Traffic direction outbound they are not drained on sigterm correctly now. This PR fixes that issue.
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: pkg/envoy/instance.go [hunk 1]
```diff
@@ -376,7 +376,7 @@ func (i *instance) ShutdownAndWait() Waitable {
 }
 
 func (i *instance) DrainListeners() error {
-	return DrainListeners(i.adminPort)
+	return DrainListeners(i.adminPort, true)
```
Comment (ramaraochavali) [inline pkg/envoy/instance.go:379]: This codepath it seems is not used any more? So I am just passing true.

**Comment by nrjpoddar** [comment]
@ramaraochavali should this be cherry-picked in release 1.6?

**Comment by ramaraochavali** [comment]
@nrjpoddar Yes. Good catch. I missed that.

**Pull request merged**